parity/migration.rs

```diff
@@ -144,26 +153,40 @@ fn migrate_database(version: u32, path: PathBuf, migrations: MigrationManager) -
 
 	println!("Migrating database {} from version {} to {}", path.to_string_lossy(), version, CURRENT_VERSION);
 
-	// get temp path
-	let temp_path = temp_database_path();
+	let temp_path = temp_database_path(&path);
+	let backup_path = backup_database_path(&path);
 	// remote the dir if it exists
 	let _ = fs::remove_dir_all(&temp_path);
+	let _ = fs::remove_dir_all(&backup_path);
```

Comment:
Agree, but it's unlikely to happen. Let's keep it as it is for now. Afaik, there is no cross platform (including windows) library for locking file. I've opened an issue not to forget about this: https://github.com/ethcore/parity/issues/1156